src/main/java/gregtech/common/covers/detector/CoverDetectorItemAdvanced.java

```diff
@@ -0,0 +1,169 @@
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
+        );
+
+        // set max fluid amount
+        group.addWidget(new LabelWidget(10, 5 + 2 * (SIZE + PADDING), "cover.advanced_item_detector.max"));
+        group.addWidget(new ImageWidget(98 - 4, 2 * (SIZE + PADDING), 4 * SIZE, SIZE, GuiTextures.DISPLAY));
+        group.addWidget(new TextFieldWidget2(98, 5 + 2 * (SIZE + PADDING), 4 * SIZE, SIZE,
+                this::getMaxValue, this::setMaxValue)
+                .setMaxLength(10)
+                .setAllowedChars(Pattern.compile(".[0-9]*"))
+        );
+
+        // invert logic button
+        group.addWidget(new LabelWidget(10, 5 + 3 * (SIZE + PADDING), "cover.advanced_energy_detector.invert_label"));
+        group.addWidget(new CycleButtonWidget(98 - 4, 3 * (SIZE + PADDING), 4 * SIZE, SIZE, this::isInverted, this::setInverted,
+                "cover.advanced_energy_detector.normal", "cover.advanced_energy_detector.inverted")
+                .setTooltipHoverString("cover.advanced_item_detector.invert_tooltip")
+        );
+
+        this.itemFilter.initUI(5 + 4 * (SIZE + PADDING), group::addWidget);
+
+        return ModularUI.builder(GuiTextures.BACKGROUND,  176, 188 + 82)
+                .widget(group)
+                .bindPlayerInventory(player.inventory, GuiTextures.SLOT, 7, 188)
+                .build(this, player);
+    }
+
+    private String getMinValue() {
+        return String.valueOf(min);
+    }
+    private String getMaxValue() {
+        return String.valueOf(max);
+    }
+    private void setMinValue(String val){
+        int parsedValue = GTUtility.tryParseInt(val, DEFAULT_MIN);
+        this.min = Math.min(max - 1, Math.max(0, parsedValue));
+    }
+    private void setMaxValue(String val){
+        int parsedValue = GTUtility.tryParseInt(val, DEFAULT_MAX);
+        max = Math.max(min + 1, parsedValue);
+    }
+
+    private boolean isInverted(){
+        return this.isInverted;
+    }
+    private void setInverted(boolean b){
+        this.isInverted = b;
+    }
+
+    @Override
+    public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, CuboidRayTraceResult hitResult) {
+        if (!this.coverHolder.getWorld().isRemote) {
+            openUI((EntityPlayerMP) playerIn);
+        }
+        return EnumActionResult.SUCCESS;
+    }
+
+    @Override
+    public void update() {
+        if (this.coverHolder.getOffsetTimer() % 20 != 0)
+            return;
+
+        IItemHandler itemHandler = coverHolder.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);
+        if (itemHandler == null)
+            return;
+
+        int storedItems = 0;
+
+        for (int i = 0; i < itemHandler.getSlots(); i++) {
+            if(itemFilter.testItemStack(itemHandler.getStackInSlot(i)))
+                storedItems += itemHandler.getStackInSlot(i).getCount();
+        }
+
+        setRedstoneSignalOutput(GTUtility.compareValue(storedItems, max, min, isInverted));
+    }
```

Comment:
Code formatting would be appreciated here as well. See my other comment about this.

Reply:
resolved with https://github.com/GregTechCEu/GregTech/pull/1448/commits/6779bc5d3151a8962d87833a1835ae2ab5571fcf